src/coreclr/utilcode/stacktrace.cpp

```diff
@@ -955,7 +955,13 @@ void MagicDeinit(void)
 *       Exactly the contents of RtlCaptureContext for Win7 - Win2K doesn't
 *       support this, so we need it for CoreCLR 4, if we require Win2K support
 ****************************************************************************/
-extern "C" __declspec(naked) void __stdcall
+extern "C"
+#ifdef TARGET_WINDOWS
+__declspec(naked)
+#else
+__attribute__((naked))
+#endif
```

Comment:
Do we need a macro for this? Like NOINLINE
Or for naked pure ASM functions, does it make sense to move the implementation to asm file?

Reply:
The only common denominator in the few usages of naked attribute is minipal, so I just inlined it. This is x86 specific hack. Most likely it can be replaced by C/C++ counterparts instead of keeping it inlined.